Trouble compiling HPX with Intel compiler

## Expected Behavior
I am trying to compile HPX on our linux-cluster  with the  Intel 19.0 compiler, however the build process fails with the same error message with any GCC version greater than 4.9.
## Actual Behavior
Build process throws following error:
`
```
remark #11074: Inlining inhibited by limit max-size 
remark #11074: Inlining inhibited by limit max-total-size 
remark #11076: To get full report use -qopt-report=4 -qopt-report-phase ipo
[ 21%] Building CXX object src/CMakeFiles/hpx.dir/runtime/threads/coroutines/detail/coroutine_impl.cpp.o
remark #11074: Inlining inhibited by limit max-total-size 
remark #11076: To get full report use -qopt-report=4 -qopt-report-phase ipo
[ 21%] Building CXX object src/CMakeFiles/hpx.dir/runtime/threads/coroutines/detail/coroutine_self.cpp.o
[ 21%] Building CXX object src/CMakeFiles/hpx.dir/runtime/threads/coroutines/detail/tss.cpp.o
[ 21%] Building CXX object src/CMakeFiles/hpx.dir/runtime/threads/coroutines/swapcontext.cpp.o
/dss/dsshome1/lrz/sys/compilers/gcc/8.2.0/bin/../include/c++/8.2.0/atomic(187): error: static assertion failed with "std::atomic requires a trivially copyable type"
        static_assert(__is_trivially_copyable(_Tp),
        ^
          detected during:
            instantiation of class "std::atomic<_Tp> [with _Tp=boost::lockfree::detail::tagged_ptr<boost::lockfree::deque_node<hpx::threads::thread_data *>>]" at line 53 of "/home/hpc/pr63so/ga84zod2/hpx/hpx/util/lockfree/deque.hpp"
            instantiation of class "boost::lockfree::deque_node<T> [with T=hpx::threads::thread_data *]" at line 151 of "/home/hpc/pr63so/ga84zod2/hpx/hpx/util/lockfree/deque.hpp"
            instantiation of class "boost::lockfree::deque<T, freelist_t, Alloc> [with T=hpx::threads::thread_data *, freelist_t=boost::lockfree::caching_freelist_t, Alloc=std::allocator<hpx::threads::thread_data *>]" at line 57 of "/home/hpc/pr63so/ga84zod2/hpx/hpx/runtime/threads/policies/lockfree_queue_backends.hpp"
            instantiation of class "hpx::threads::policies::lockfree_fifo_backend<T> [with T=hpx::threads::thread_data *]" at line 1153 of "/home/hpc/pr63so/ga84zod2/hpx/hpx/runtime/threads/policies/thread_queue.hpp"
            instantiation of class "hpx::threads::policies::thread_queue<Mutex, PendingQueuing, StagedQueuing, TerminatedQueuing> [with Mutex=hpx::compat::mutex, PendingQueuing=hpx::threads::policies::lockfree_fifo, StagedQueuing=hpx::threads::policies::lockfree_fifo, TerminatedQueuing=hpx::threads::policies::lockfree_lifo]" at line 1288 of "/home/hpc/pr63so/ga84zod2/hpx/hpx/runtime/threads/policies/local_priority_queue_scheduler.hpp"
            instantiation of class "hpx::threads::policies::local_priority_queue_scheduler<Mutex, PendingQueuing, StagedQueuing, TerminatedQueuing> [with Mutex=hpx::compat::mutex, PendingQueuing=hpx::threads::policies::lockfree_fifo, StagedQueuing=hpx::threads::policies::lockfree_fifo, TerminatedQueuing=hpx::threads::policies::lockfree_lifo]" at line 43 of "/home/hpc/pr63so/ga84zod2/hpx/hpx/runtime/threads/policies/static_priority_queue_scheduler.hpp"
            instantiation of class "hpx::threads::policies::static_priority_queue_scheduler<Mutex, PendingQueuing, StagedQueuing, TerminatedQueuing> [with Mutex=hpx::compat::mutex, PendingQueuing=hpx::threads::policies::lockfree_fifo, StagedQueuing=hpx::threads::policies::lockfree_fifo, TerminatedQueuing=hpx::threads::policies::lockfree_lifo]" at line 27 of "/home/hpc/pr63so/ga84zod2/hpx/src/runtime/threads/detail/scheduled_thread_pool.cpp"

/dss/dsshome1/lrz/sys/compilers/gcc/8.2.0/bin/../include/c++/8.2.0/atomic(187): error: static assertion failed with "std::atomic requires a trivially copyable type"
        static_assert(__is_trivially_copyable(_Tp),
        ^oost::lockfree::detail::tagged_ptr<boost::lockfree::deque_node<hpx::util::tuple<hpx::threads::thread_init_data, hpx::threads::thread_state_enum> *>>]" at line 53 of "/home/hpc/pr63so/ga84zod2/hpx/hpx/util/lockfree/deque.hpp"
            instantiation of class "boost::lockfree::deque_node<T> [with T=hpx::util::tuple<hpx::threads::thread_init_data, hpx::threads::thread_state_enum> *]" at line 151 of "/home/hpc/pr63so/ga84zod2/hpx/hpx/util/lockfree/deque.hpp"
            instantiation of class "boost::lockfree::deque<T, freelist_t, Alloc> [with T=hpx::util::tuple<hpx::threads::thread_init_data, hpx::threads::thread_state_enum> *, freelist_t=boost::lockfree::caching_freelist_t, Alloc=std::allocator<hpx::util::tuple<hpx::threads::thread_init_data, hpx::threads::thread_state_enum> *>]" at line 57 of "/home/hpc/pr63so/ga84zod2/hpx/hpx/runtime/threads/policies/lockfree_queue_backends.hpp"
            instantiation of class "hpx::threads::policies::lockfree_fifo_backend<T> [with T=hpx::util::tuple<hpx::threads::thread_init_data, hpx::threads::thread_state_enum> *]" at line 1164 of "/home/hpc/pr63so/ga84zod2/hpx/hpx/runtime/threads/policies/thread_queue.hpp"
            instantiation of class "hpx::threads::policies::thread_queue<Mutex, PendingQueuing, StagedQueuing, TerminatedQueuing> [with Mutex=hpx::compat::mutex, PendingQueuing=hpx::threads::policies::lockfree_fifo, StagedQueuing=hpx::threads::policies::lockfree_fifo, TerminatedQueuing=hpx::threads::policies::lockfree_lifo]" at line 1288 of "/home/hpc/pr63so/ga84zod2/hpx/hpx/runtime/threads/policies/local_priority_queue_scheduler.hpp"
            instantiation of class "hpx::threads::policies::local_priority_queue_scheduler<Mutex, PendingQueuing, StagedQueuing, TerminatedQueuing> [with Mutex=hpx::compat::mutex, PendingQueuing=hpx::threads::policies::lockfree_fifo, StagedQueuing=hpx::threads::policies::lockfree_fifo, TerminatedQueuing=hpx::threads::policies::lockfree_lifo]" at line 43 of "/home/hpc/pr63so/ga84zod2/hpx/hpx/runtime/threads/policies/static_priority_queue_scheduler.hpp"
            instantiation of class "hpx::threads::policies::static_priority_queue_scheduler<Mutex, PendingQueuing, StagedQueuing, TerminatedQueuing> [with Mutex=hpx::compat::mutex, PendingQueuing=hpx::threads::policies::lockfree_fifo, StagedQueuing=hpx::threads::policies::lockfree_fifo, TerminatedQueuing=hpx::threads::policies::lockfree_lifo]" at line 27 of "/home/hpc/pr63so/ga84zod2/hpx/src/runtime/threads/detail/scheduled_thread_pool.cpp"

/home/hpc/pr63so/ga84zod2/hpx/src/runtime/threads/detail/scheduled_thread_pool.cpp(33): warning #1286: invalid attribute for "hpx::threads::policies::local_priority_queue_scheduler<hpx::compat::mutex, hpx::threads::policies::lockfree_fifo, hpx::threads::policies::lockfree_fifo, hpx::threads::policies::lockfree_lifo>"
  template class HPX_EXPORT hpx::threads::policies::local_priority_queue_scheduler<
                 ^

[ 21%] Building CXX object src/CMakeFiles/hpx.dir/runtime/threads/executors/current_executor.cpp.o
[ 21%] Building CXX object src/CMakeFiles/hpx.dir/runtime/threads/executors/default_executor.cpp.o
[ 21%] Building CXX object src/CMakeFiles/hpx.dir/runtime/threads/executors/pool_executor.cpp.o
remark #11074: Inlining inhibited by limit max-size 
remark #11074: Inlining inhibited by limit max-total-size 
remark #11076: To get full report use -qopt-report=4 -qopt-report-phase ipo
[ 21%] Building CXX object src/CMakeFiles/hpx.dir/runtime/threads/executors/service_executor.cpp.o
[ 21%] Building CXX object src/CMakeFiles/hpx.dir/runtime/threads/executors/this_thread_executors.cpp.o
remark #11074: Inlining inhibited by limit max-total-size 
remark #11076: To get full report use -qopt-report=4 -qopt-report-phase ipo
In file included from /home/hpc/pr63so/ga84zod2/hpx/hpx/parallel/executors/execution.hpp(21),
                 from /home/hpc/pr63so/ga84zod2/hpx/hpx/lcos/local/packaged_continuation.hpp(23),
                 from /home/hpc/pr63so/ga84zod2/hpx/hpx/lcos/future.hpp(1715),
                 from /home/hpc/pr63so/ga84zod2/hpx/hpx/runtime/threads/thread_pool_base.hpp(16),
                 from /home/hpc/pr63so/ga84zod2/hpx/hpx/runtime/threads/executors/pool_executor.hpp(12),
                 from /home/hpc/pr63so/ga84zod2/hpx/src/runtime/threads/executors/pool_executor.cpp(7):
/home/hpc/pr63so/ga84zod2/hpx/hpx/lcos/wait_all.hpp(382): warning #3058: GNU attributes on a template redeclaration have no effect
      HPX_FORCEINLINE void wait_all(std::vector<Future>&& values)
      ^

remark #11074: Inlining inhibited by limit max-size 
remark #11074: Inlining inhibited by limit max-total-size 
remark #11076: To get full report use -qopt-report=4 -qopt-report-phase ipo
In file included from /home/hpc/pr63so/ga84zod2/hpx/hpx/parallel/executors/execution.hpp(21),
                 from /home/hpc/pr63so/ga84zod2/hpx/hpx/lcos/local/packaged_continuation.hpp(23),
                 from /home/hpc/pr63so/ga84zod2/hpx/hpx/lcos/future.hpp(1715),
                 from /home/hpc/pr63so/ga84zod2/hpx/hpx/runtime/components/make_client.hpp(10),
                 from /home/hpc/pr63so/ga84zod2/hpx/hpx/runtime/basename_registration_fwd.hpp(15),
                 from /home/hpc/pr63so/ga84zod2/hpx/hpx/runtime_fwd.hpp(14),
                 from /home/hpc/pr63so/ga84zod2/hpx/src/runtime/threads/executors/service_executor.cpp(11):
/home/hpc/pr63so/ga84zod2/hpx/hpx/lcos/wait_all.hpp(382): warning #3058: GNU attributes on a template redeclaration have no effect
      HPX_FORCEINLINE void wait_all(std::vector<Future>&& values)
      ^

In file included from /home/hpc/pr63so/ga84zod2/hpx/hpx/parallel/executors/execution.hpp(21),
                 from /home/hpc/pr63so/ga84zod2/hpx/hpx/lcos/local/packaged_continuation.hpp(23),
                 from /home/hpc/pr63so/ga84zod2/hpx/hpx/lcos/future.hpp(1715),
                 from /home/hpc/pr63so/ga84zod2/hpx/hpx/runtime/threads/thread_pool_base.hpp(16),
                 from /home/hpc/pr63so/ga84zod2/hpx/hpx/runtime/threads/policies/scheduler_base.hpp(15),
                 from /home/hpc/pr63so/ga84zod2/hpx/hpx/runtime/threads/detail/create_thread.hpp(10),
                 from /home/hpc/pr63so/ga84zod2/hpx/src/runtime/threads/executors/current_executor.cpp(12):
/home/hpc/pr63so/ga84zod2/hpx/hpx/lcos/wait_all.hpp(382): warning #3058: GNU attributes on a template redeclaration have no effect
      HPX_FORCEINLINE void wait_all(std::vector<Future>&& values)
      ^

[ 21%] Building CXX object src/CMakeFiles/hpx.dir/runtime/threads/executors/thread_pool_attached_executors.cpp.o
remark #11074: Inlining inhibited by limit max-size 
remark #11074: Inlining inhibited by limit max-total-size 
remark #11076: To get full report use -qopt-report=4 -qopt-report-phase ipo
[ 21%] Building CXX object src/CMakeFiles/hpx.dir/runtime/threads/executors/thread_pool_executors.cpp.o
In file included from /home/hpc/pr63so/ga84zod2/hpx/hpx/parallel/executors/execution.hpp(21),
                 from /home/hpc/pr63so/ga84zod2/hpx/hpx/lcos/local/packaged_continuation.hpp(23),
                 from /home/hpc/pr63so/ga84zod2/hpx/hpx/lcos/future.hpp(1715),
                 from /home/hpc/pr63so/ga84zod2/hpx/hpx/runtime/threads/thread_pool_base.hpp(16),
                 from /home/hpc/pr63so/ga84zod2/hpx/hpx/runtime/threads/policies/scheduler_base.hpp(15),
                 from /home/hpc/pr63so/ga84zod2/hpx/hpx/runtime/threads/policies/local_priority_queue_scheduler.hpp(13),
                 from /home/hpc/pr63so/ga84zod2/hpx/hpx/runtime/threads/policies/static_priority_queue_scheduler.hpp(16),
                 from /home/hpc/pr63so/ga84zod2/hpx/src/runtime/threads/executors/this_thread_executors.cpp(11):
/home/hpc/pr63so/ga84zod2/hpx/hpx/lcos/wait_all.hpp(382): warning #3058: GNU attributes on a template redeclaration have no effect
      HPX_FORCEINLINE void wait_all(std::vector<Future>&& values)
      ^

/dss/dsshome1/lrz/sys/compilers/gcc/8.2.0/bin/../include/c++/8.2.0/atomic(187): error: static assertion failed with "std::atomic requires a trivially copyable type"
        static_assert(__is_trivially_copyable(_Tp),
        ^
          detected during:
            instantiation of class "std::atomic<_Tp> [with _Tp=boost::lockfree::detail::tagged_ptr<boost::lockfree::deque_node<hpx::threads::thread_data *>>]" at line 53 of "/home/hpc/pr63so/ga84zod2/hpx/hpx/util/lockfree/deque.hpp"
            instantiation of class "boost::lockfree::deque_node<T> [with T=hpx::threads::thread_data *]" at line 151 of "/home/hpc/pr63so/ga84zod2/hpx/hpx/util/lockfree/deque.hpp"
            instantiation of class "boost::lockfree::deque<T, freelist_t, Alloc> [with T=hpx::threads::thread_data *, freelist_t=boost::lockfree::caching_freelist_t, Alloc=std::allocator<hpx::threads::thread_data *>]" at line 57 of "/home/hpc/pr63so/ga84zod2/hpx/hpx/runtime/threads/policies/lockfree_queue_backends.hpp"
            instantiation of class "hpx::threads::policies::lockfree_fifo_backend<T> [with T=hpx::threads::thread_data *]" at line 1153 of "/home/hpc/pr63so/ga84zod2/hpx/hpx/runtime/threads/policies/thread_queue.hpp"
            instantiation of class "hpx::threads::policies::thread_queue<Mutex, PendingQueuing, StagedQueuing, TerminatedQueuing> [with Mutex=hpx::compat::mutex, PendingQueuing=hpx::threads::policies::lockfree_fifo, StagedQueuing=hpx::threads::policies::lockfree_fifo, TerminatedQueuing=hpx::threads::policies::lockfree_lifo]" at line 1288 of "/home/hpc/pr63so/ga84zod2/hpx/hpx/runtime/threads/policies/local_priority_queue_scheduler.hpp"
            instantiation of class "hpx::threads::policies::local_priority_queue_scheduler<Mutex, PendingQueuing, StagedQueuing, TerminatedQueuing> [with Mutex=hpx::compat::mutex, PendingQueuing=hpx::threads::policies::lockfree_fifo, StagedQueuing=hpx::threads::policies::lockfree_fifo, TerminatedQueuing=hpx::threads::policies::lockfree_lifo]" at line 43 of "/home/hpc/pr63so/ga84zod2/hpx/hpx/runtime/threads/policies/static_priority_queue_scheduler.hpp"
            instantiation of class "hpx::threads::policies::static_priority_queue_scheduler<Mutex, PendingQueuing, StagedQueuing, TerminatedQueuing> [with Mutex=hpx::compat::mutex, PendingQueuing=hpx::threads::policies::lockfree_fifo, StagedQueuing=hpx::threads::policies::lockfree_fifo, TerminatedQueuing=hpx::threads::policies::lockfree_lifo]" at line 116 of "/home/hpc/pr63so/ga84zod2/hpx/hpx/runtime/threads/executors/this_thread_executors.hpp"
            instantiation of class "hpx::threads::executors::detail::this_thread_executor<Scheduler> [with Scheduler=hpx::threads::policies::static_priority_queue_scheduler<hpx::compat::mutex, hpx::threads::policies::lockfree_fifo, hpx::threads::policies::lockfree_fifo, hpx::threads::policies::lockfree_lifo>]" at line 483 of "/home/hpc/pr63so/ga84zod2/hpx/src/runtime/threads/executors/this_thread_executors.cpp"

/dss/dsshome1/lrz/sys/compilers/gcc/8.2.0/bin/../include/c++/8.2.0/atomic(187): error: static assertion failed with "std::atomic requires a trivially copyable type"
        static_assert(__is_trivially_copyable(_Tp),
        ^
          detected during:
            instantiation of class "std::atomic<_Tp> [with _Tp=boost::lockfree::detail::tagged_ptr<boost::lockfree::deque_node<hpx::util::tuple<hpx::threads::thread_init_data, hpx::threads::thread_state_enum> *>>]" at line 53 of "/home/hpc/pr63so/ga84zod2/hpx/hpx/util/lockfree/deque.hpp"
            instantiation of class "boost::lockfree::deque_node<T> [with T=hpx::util::tuple<hpx::threads::thread_init_data, hpx::threads::thread_state_enum> *]" at line 151 of "/home/hpc/pr63so/ga84zod2/hpx/hpx/util/lockfree/deque.hpp"
            instantiation of class "boost::lockfree::deque<T, freelist_t, Alloc> [with T=hpx::util::tuple<hpx::threads::thread_init_data, hpx::threads::thread_state_enum> *, freelist_t=boost::lockfree::caching_freelist_t, Alloc=std::allocator<hpx::util::tuple<hpx::threads::thread_init_data, hpx::threads::thread_state_enum> *>]" at line 57 of "/home/hpc/pr63so/ga84zod2/hpx/hpx/runtime/threads/policies/lockfree_queue_backends.hpp"
            instantiation of class "hpx::threads::policies::lockfree_fifo_backend<T> [with T=hpx::util::tuple<hpx::threads::thread_init_data, hpx::threads::thread_state_enum> *]" at line 1164 of "/home/hpc/pr63so/ga84zod2/hpx/hpx/runtime/threads/policies/thread_queue.hpp"
            instantiation of class "hpx::threads::policies::thread_queue<Mutex, PendingQueuing, StagedQueuing, TerminatedQueuing> [with Mutex=hpx::compat::mutex, PendingQueuing=hpx::threads::policies::lockfree_fifo, StagedQueuing=hpx::threads::policies::lockfree_fifo, TerminatedQueuing=hpx::threads::policies::lockfree_lifo]" at line 1288 of "/home/hpc/pr63so/ga84zod2/hpx/hpx/runtime/threads/policies/local_priority_queue_scheduler.hpp"
            instantiation of class "hpx::threads::policies::local_priority_queue_scheduler<Mutex, PendingQueuing, StagedQueuing, TerminatedQueuing> [with Mutex=hpx::compat::mutex, PendingQueuing=hpx::threads::policies::lockfree_fifo, StagedQueuing=hpx::threads::policies::lockfree_fifo, TerminatedQueuing=hpx::threads::policies::lockfree_lifo]" at line 43 of "/home/hpc/pr63so/ga84zod2/hpx/hpx/runtime/threads/policies/static_priority_queue_scheduler.hpp"
            instantiation of class "hpx::threads::policies::static_priority_queue_scheduler<Mutex, PendingQueuing, StagedQueuing, TerminatedQueuing> [with Mutex=hpx::compat::mutex, PendingQueuing=hpx::threads::policies::lockfree_fifo, StagedQueuing=hpx::threads::policies::lockfree_fifo, TerminatedQueuing=hpx::threads::policies::lockfree_lifo]" at line 116 of "/home/hpc/pr63so/ga84zod2/hpx/hpx/runtime/threads/executors/this_thread_executors.hpp"
            instantiation of class "hpx::threads::executors::detail::this_thread_executor<Scheduler> [with Scheduler=hpx::threads::policies::static_priority_queue_scheduler<hpx::compat::mutex, hpx::threads::policies::lockfree_fifo, hpx::threads::policies::lockfree_fifo, hpx::threads::policies::lockfree_lifo>]" at line 483 of "/home/hpc/pr63so/ga84zod2/hpx/src/runtime/threads/executors/this_thread_executors.cpp"

compilation aborted for /home/hpc/pr63so/ga84zod2/hpx/src/runtime/threads/detail/scheduled_thread_pool.cpp (code 2)
src/CMakeFiles/hpx.dir/build.make:1622: recipe for target 'src/CMakeFiles/hpx.dir/runtime/threads/detail/scheduled_thread_pool.cpp.o' failed
make[2]: *** [src/CMakeFiles/hpx.dir/runtime/threads/detail/scheduled_thread_pool.cpp.o] Error 2
make[2]: *** Waiting for unfinished jobs....
In file included from /home/hpc/pr63so/ga84zod2/hpx/hpx/parallel/executors/execution.hpp(21),
                 from /home/hpc/pr63so/ga84zod2/hpx/hpx/lcos/local/packaged_continuation.hpp(23),
                 from /home/hpc/pr63so/ga84zod2/hpx/hpx/lcos/future.hpp(1715),
                 from /home/hpc/pr63so/ga84zod2/hpx/hpx/runtime/threads/thread_pool_base.hpp(16),
                 from /home/hpc/pr63so/ga84zod2/hpx/hpx/runtime/threads/policies/scheduler_base.hpp(15),
                 from /home/hpc/pr63so/ga84zod2/hpx/hpx/runtime/threads/policies/local_queue_scheduler.hpp(15),
                 from /home/hpc/pr63so/ga84zod2/hpx/src/runtime/threads/executors/thread_pool_attached_executors.cpp(10):
/home/hpc/pr63so/ga84zod2/hpx/hpx/lcos/wait_all.hpp(382): warning #3058: GNU attributes on a template redeclaration have no effect
      HPX_FORCEINLINE void wait_all(std::vector<Future>&& values)
      ^

remark #11074: Inlining inhibited by limit max-size 
remark #11074: Inlining inhibited by limit max-total-size 
remark #11076: To get full report use -qopt-report=4 -qopt-report-phase ipo
compilation aborted for /home/hpc/pr63so/ga84zod2/hpx/src/runtime/threads/executors/this_thread_executors.cpp (code 2)
src/CMakeFiles/hpx.dir/build.make:1886: recipe for target 'src/CMakeFiles/hpx.dir/runtime/threads/executors/this_thread_executors.cpp.o' failed
make[2]: *** [src/CMakeFiles/hpx.dir/runtime/threads/executors/this_thread_executors.cpp.o] Error 2
In file included from /home/hpc/pr63so/ga84zod2/hpx/hpx/parallel/executors/execution.hpp(21),
                 from /home/hpc/pr63so/ga84zod2/hpx/hpx/lcos/local/packaged_continuation.hpp(23),
                 from /home/hpc/pr63so/ga84zod2/hpx/hpx/lcos/future.hpp(1715),
                 from /home/hpc/pr63so/ga84zod2/hpx/hpx/runtime/threads/thread_pool_base.hpp(16),
                 from /home/hpc/pr63so/ga84zod2/hpx/hpx/runtime/threads/policies/scheduler_base.hpp(15),
                 from /home/hpc/pr63so/ga84zod2/hpx/hpx/runtime/threads/policies/local_queue_scheduler.hpp(15),
                 from /home/hpc/pr63so/ga84zod2/hpx/src/runtime/threads/executors/thread_pool_executors.cpp(10):
/home/hpc/pr63so/ga84zod2/hpx/hpx/lcos/wait_all.hpp(382): warning #3058: GNU attributes on a template redeclaration have no effect
      HPX_FORCEINLINE void wait_all(std::vector<Future>&& values)
      ^

/dss/dsshome1/lrz/sys/compilers/gcc/8.2.0/bin/../include/c++/8.2.0/atomic(187): error: static assertion failed with "std::atomic requires a trivially copyable type"
        static_assert(__is_trivially_copyable(_Tp),
        ^
          detected during:
            instantiation of class "std::atomic<_Tp> [with _Tp=boost::lockfree::detail::tagged_ptr<boost::lockfree::deque_node<hpx::threads::thread_data *>>]" at line 53 of "/home/hpc/pr63so/ga84zod2/hpx/hpx/util/lockfree/deque.hpp"
            instantiation of class "boost::lockfree::deque_node<T> [with T=hpx::threads::thread_data *]" at line 151 of "/home/hpc/pr63so/ga84zod2/hpx/hpx/util/lockfree/deque.hpp"
            instantiation of class "boost::lockfree::deque<T, freelist_t, Alloc> [with T=hpx::threads::thread_data *, freelist_t=boost::lockfree::caching_freelist_t, Alloc=std::allocator<hpx::threads::thread_data *>]" at line 57 of "/home/hpc/pr63so/ga84zod2/hpx/hpx/runtime/threads/policies/lockfree_queue_backends.hpp"
            instantiation of class "hpx::threads::policies::lockfree_fifo_backend<T> [with T=hpx::threads::thread_data *]" at line 1153 of "/home/hpc/pr63so/ga84zod2/hpx/hpx/runtime/threads/policies/thread_queue.hpp"
            instantiation of class "hpx::threads::policies::thread_queue<Mutex, PendingQueuing, StagedQueuing, TerminatedQueuing> [with Mutex=hpx::compat::mutex, PendingQueuing=hpx::threads::policies::lockfree_fifo, StagedQueuing=hpx::threads::policies::lockfree_fifo, TerminatedQueuing=hpx::threads::policies::lockfree_lifo]" at line 1288 of "/home/hpc/pr63so/ga84zod2/hpx/hpx/runtime/threads/policies/local_priority_queue_scheduler.hpp"
            instantiation of class "hpx::threads::policies::local_priority_queue_scheduler<Mutex, PendingQueuing, StagedQueuing, TerminatedQueuing> [with Mutex=hpx::compat::mutex, PendingQueuing=hpx::threads::policies::lockfree_fifo, StagedQueuing=hpx::threads::policies::lockfree_fifo, TerminatedQueuing=hpx::threads::policies::lockfree_lifo]" at line 127 of "/home/hpc/pr63so/ga84zod2/hpx/hpx/runtime/threads/executors/thread_pool_executors.hpp"
            instantiation of class "hpx::threads::executors::detail::thread_pool_executor<Scheduler> [with Scheduler=hpx::threads::policies::local_priority_queue_scheduler<hpx::compat::mutex, hpx::threads::policies::lockfree_fifo, hpx::threads::policies::lockfree_fifo, hpx::threads::policies::lockfree_lifo>]" at line 500 of "/home/hpc/pr63so/ga84zod2/hpx/src/runtime/threads/executors/thread_pool_executors.cpp"

/dss/dsshome1/lrz/sys/compilers/gcc/8.2.0/bin/../include/c++/8.2.0/atomic(187): error: static assertion failed with "std::atomic requires a trivially copyable type"
        static_assert(__is_trivially_copyable(_Tp),
        ^
          detected during:
            instantiation of class "std::atomic<_Tp> [with _Tp=boost::lockfree::detail::tagged_ptr<boost::lockfree::deque_node<hpx::util::tuple<hpx::threads::thread_init_data, hpx::threads::thread_state_enum> *>>]" at line 53 of "/home/hpc/pr63so/ga84zod2/hpx/hpx/util/lockfree/deque.hpp"
            instantiation of class "boost::lockfree::deque_node<T> [with T=hpx::util::tuple<hpx::threads::thread_init_data, hpx::threads::thread_state_enum> *]" at line 151 of "/home/hpc/pr63so/ga84zod2/hpx/hpx/util/lockfree/deque.hpp"
            instantiation of class "boost::lockfree::deque<T, freelist_t, Alloc> [with T=hpx::util::tuple<hpx::threads::thread_init_data, hpx::threads::thread_state_enum> *, freelist_t=boost::lockfree::caching_freelist_t, Alloc=std::allocator<hpx::util::tuple<hpx::threads::thread_init_data, hpx::threads::thread_state_enum> *>]" at line 57 of "/home/hpc/pr63so/ga84zod2/hpx/hpx/runtime/threads/policies/lockfree_queue_backends.hpp"
            instantiation of class "hpx::threads::policies::lockfree_fifo_backend<T> [with T=hpx::util::tuple<hpx::threads::thread_init_data, hpx::threads::thread_state_enum> *]" at line 1164 of "/home/hpc/pr63so/ga84zod2/hpx/hpx/runtime/threads/policies/thread_queue.hpp"
            instantiation of class "hpx::threads::policies::thread_queue<Mutex, PendingQueuing, StagedQueuing, TerminatedQueuing> [with Mutex=hpx::compat::mutex, PendingQueuing=hpx::threads::policies::lockfree_fifo, StagedQueuing=hpx::threads::policies::lockfree_fifo, TerminatedQueuing=hpx::threads::policies::lockfree_lifo]" at line 1288 of "/home/hpc/pr63so/ga84zod2/hpx/hpx/runtime/threads/policies/local_priority_queue_scheduler.hpp"
            instantiation of class "hpx::threads::policies::local_priority_queue_scheduler<Mutex, PendingQueuing, StagedQueuing, TerminatedQueuing> [with Mutex=hpx::compat::mutex, PendingQueuing=hpx::threads::policies::lockfree_fifo, StagedQueuing=hpx::threads::policies::lockfree_fifo, TerminatedQueuing=hpx::threads::policies::lockfree_lifo]" at line 127 of "/home/hpc/pr63so/ga84zod2/hpx/hpx/runtime/threads/executors/thread_pool_executors.hpp"
            instantiation of class "hpx::threads::executors::detail::thread_pool_executor<Scheduler> [with Scheduler=hpx::threads::policies::local_priority_queue_scheduler<hpx::compat::mutex, hpx::threads::policies::lockfree_fifo, hpx::threads::policies::lockfree_fifo, hpx::threads::policies::lockfree_lifo>]" at line 500 of "/home/hpc/pr63so/ga84zod2/hpx/src/runtime/threads/executors/thread_pool_executors.cpp"

compilation aborted for /home/hpc/pr63so/ga84zod2/hpx/src/runtime/threads/executors/thread_pool_executors.cpp (code 2)
src/CMakeFiles/hpx.dir/build.make:1934: recipe for target 'src/CMakeFiles/hpx.dir/runtime/threads/executors/thread_pool_executors.cpp.o' failed
make[2]: *** [src/CMakeFiles/hpx.dir/runtime/threads/executors/thread_pool_executors.cpp.o] Error 2
          detected during:
            instantiation of class "std::atomic<_Tp> [with _Tp=boost::lockfree::detail::tagged_ptr<boost::lockfree::deque_node<hpx::util::tuple<hpx::threads::thread_init_data, hpx::threads::thread_state_enum> *>>]" at line 53 of "/home/hpc/pr63so/ga84zod2/hpx/hpx/util/lockfree/deque.hpp"
            instantiation of class "boost::lockfree::deque_node<T> [with T=hpx::util::tuple<hpx::threads::thread_init_data, hpx::threads::thread_state_enum> *]" at line 151 of "/home/hpc/pr63so/ga84zod2/hpx/hpx/util/lockfree/deque.hpp"
            instantiation of class "boost::lockfree::deque<T, freelist_t, Alloc> [with T=hpx::util::tuple<hpx::threads::thread_init_data, hpx::threads::thread_state_enum> *, freelist_t=boost::lockfree::caching_freelist_t, Alloc=std::allocator<hpx::util::tuple<hpx::threads::thread_init_data, hpx::threads::thread_state_enum> *>]" at line 57 of "/home/hpc/pr63so/ga84zod2/hpx/hpx/runtime/threads/policies/lockfree_queue_backends.hpp"
            instantiation of class "hpx::threads::policies::lockfree_fifo_backend<T> [with T=hpx::util::tuple<hpx::threads::thread_init_data, hpx::threads::thread_state_enum> *]" at line 1164 of "/home/hpc/pr63so/ga84zod2/hpx/hpx/runtime/threads/policies/thread_queue.hpp"
            instantiation of class "hpx::threads::policies::thread_queue<Mutex, PendingQueuing, StagedQueuing, TerminatedQueuing> [with Mutex=hpx::compat::mutex, PendingQueuing=hpx::threads::policies::lockfree_fifo, StagedQueuing=hpx::threads::policies::lockfree_fifo, TerminatedQueuing=hpx::threads::policies::lockfree_lifo]" at line 1288 of "/home/hpc/pr63so/ga84zod2/hpx/hpx/runtime/threads/policies/local_priority_queue_scheduler.hpp"
            instantiation of class "hpx::threads::policies::local_priority_queue_scheduler<Mutex, PendingQueuing, StagedQueuing, TerminatedQueuing> [with Mutex=hpx::compat::mutex, PendingQueuing=hpx::threads::policies::lockfree_fifo, StagedQueuing=hpx::threads::policies::lockfree_fifo, TerminatedQueuing=hpx::threads::policies::lockfree_lifo]" at line 43 of "/home/hpc/pr63so/ga84zod2/hpx/hpx/runtime/threads/policies/static_priority_queue_scheduler.hpp"
            instantiation of class "hpx::threads::policies::static_priority_queue_scheduler<Mutex, PendingQueuing, StagedQueuing, TerminatedQueuing> [with Mutex=hpx::compat::mutex, PendingQueuing=hpx::threads::policies::lockfree_fifo, StagedQueuing=hpx::threads::policies::lockfree_fifo, TerminatedQueuing=hpx::threads::policies::lockfree_lifo]" at line 27 of "/home/hpc/pr63so/ga84zod2/hpx/src/runtime/threads/detail/scheduled_thread_pool.cpp"

/home/hpc/pr63so/ga84zod2/hpx/src/runtime/threads/detail/scheduled_thread_pool.cpp(33): warning #1286: invalid attribute for "hpx::threads::policies::local_priority_queue_scheduler<hpx::compat::mutex, hpx::threads::policies::lockfree_fifo, hpx::threads::policies::lockfree_fifo, hpx::threads::policies::lockfree_lifo>"
  template class HPX_EXPORT hpx::threads::policies::local_priority_queue_scheduler<
                 ^

[ 21%] Building CXX object src/CMakeFiles/hpx.dir/runtime/threads/executors/current_executor.cpp.o
[ 21%] Building CXX object src/CMakeFiles/hpx.dir/runtime/threads/executors/default_executor.cpp.o
[ 21%] Building CXX object src/CMakeFiles/hpx.dir/runtime/threads/executors/pool_executor.cpp.o
remark #11074: Inlining inhibited by limit max-size 
remark #11074: Inlining inhibited by limit max-total-size 
remark #11076: To get full report use -qopt-report=4 -qopt-report-phase ipo
[ 21%] Building CXX object src/CMakeFiles/hpx.dir/runtime/threads/executors/service_executor.cpp.o
[ 21%] Building CXX object src/CMakeFiles/hpx.dir/runtime/threads/executors/this_thread_executors.cpp.o
remark #11074: Inlining inhibited by limit max-total-size 
remark #11076: To get full report use -qopt-report=4 -qopt-report-phase ipo
In file included from /home/hpc/pr63so/ga84zod2/hpx/hpx/parallel/executors/execution.hpp(21),
                 from /home/hpc/pr63so/ga84zod2/hpx/hpx/lcos/local/packaged_continuation.hpp(23),
                 from /home/hpc/pr63so/ga84zod2/hpx/hpx/lcos/future.hpp(1715),
                 from /home/hpc/pr63so/ga84zod2/hpx/hpx/runtime/threads/thread_pool_base.hpp(16),
                 from /home/hpc/pr63so/ga84zod2/hpx/hpx/runtime/threads/executors/pool_executor.hpp(12),
                 from /home/hpc/pr63so/ga84zod2/hpx/src/runtime/threads/executors/pool_executor.cpp(7):
/home/hpc/pr63so/ga84zod2/hpx/hpx/lcos/wait_all.hpp(382): warning #3058: GNU attributes on a template redeclaration have no effect
      HPX_FORCEINLINE void wait_all(std::vector<Future>&& values)
      ^

remark #11074: Inlining inhibited by limit max-size 
remark #11074: Inlining inhibited by limit max-total-size 
remark #11076: To get full report use -qopt-report=4 -qopt-report-phase ipo
In file included from /home/hpc/pr63so/ga84zod2/hpx/hpx/parallel/executors/execution.hpp(21),
                 from /home/hpc/pr63so/ga84zod2/hpx/hpx/lcos/local/packaged_continuation.hpp(23),
                 from /home/hpc/pr63so/ga84zod2/hpx/hpx/lcos/future.hpp(1715),
                 from /home/hpc/pr63so/ga84zod2/hpx/hpx/runtime/components/make_client.hpp(10),
                 from /home/hpc/pr63so/ga84zod2/hpx/hpx/runtime/basename_registration_fwd.hpp(15),
                 from /home/hpc/pr63so/ga84zod2/hpx/hpx/runtime_fwd.hpp(14),
                 from /home/hpc/pr63so/ga84zod2/hpx/src/runtime/threads/executors/service_executor.cpp(11):
/home/hpc/pr63so/ga84zod2/hpx/hpx/lcos/wait_all.hpp(382): warning #3058: GNU attributes on a template redeclaration have no effect
      HPX_FORCEINLINE void wait_all(std::vector<Future>&& values)
      ^

In file included from /home/hpc/pr63so/ga84zod2/hpx/hpx/parallel/executors/execution.hpp(21),
                 from /home/hpc/pr63so/ga84zod2/hpx/hpx/lcos/local/packaged_continuation.hpp(23),
                 from /home/hpc/pr63so/ga84zod2/hpx/hpx/lcos/future.hpp(1715),
                 from /home/hpc/pr63so/ga84zod2/hpx/hpx/runtime/threads/thread_pool_base.hpp(16),
                 from /home/hpc/pr63so/ga84zod2/hpx/hpx/runtime/threads/policies/scheduler_base.hpp(15),
                 from /home/hpc/pr63so/ga84zod2/hpx/hpx/runtime/threads/detail/create_thread.hpp(10),
                 from /home/hpc/pr63so/ga84zod2/hpx/src/runtime/threads/executors/current_executor.cpp(12):
/home/hpc/pr63so/ga84zod2/hpx/hpx/lcos/wait_all.hpp(382): warning #3058: GNU attributes on a template redeclaration have no effect
      HPX_FORCEINLINE void wait_all(std::vector<Future>&& values)
      ^

[ 21%] Building CXX object src/CMakeFiles/hpx.dir/runtime/threads/executors/thread_pool_attached_executors.cpp.o
remark #11074: Inlining inhibited by limit max-size 
remark #11074: Inlining inhibited by limit max-total-size 
remark #11076: To get full report use -qopt-report=4 -qopt-report-phase ipo
[ 21%] Building CXX object src/CMakeFiles/hpx.dir/runtime/threads/executors/thread_pool_executors.cpp.o
In file included from /home/hpc/pr63so/ga84zod2/hpx/hpx/parallel/executors/execution.hpp(21),
                 from /home/hpc/pr63so/ga84zod2/hpx/hpx/lcos/local/packaged_continuation.hpp(23),
                 from /home/hpc/pr63so/ga84zod2/hpx/hpx/lcos/future.hpp(1715),
                 from /home/hpc/pr63so/ga84zod2/hpx/hpx/runtime/threads/thread_pool_base.hpp(16),
                 from /home/hpc/pr63so/ga84zod2/hpx/hpx/runtime/threads/policies/scheduler_base.hpp(15),
                 from /home/hpc/pr63so/ga84zod2/hpx/hpx/runtime/threads/policies/local_priority_queue_scheduler.hpp(13),
                 from /home/hpc/pr63so/ga84zod2/hpx/hpx/runtime/threads/policies/static_priority_queue_scheduler.hpp(16),
                 from /home/hpc/pr63so/ga84zod2/hpx/src/runtime/threads/executors/this_thread_executors.cpp(11):
/home/hpc/pr63so/ga84zod2/hpx/hpx/lcos/wait_all.hpp(382): warning #3058: GNU attributes on a template redeclaration have no effect
      HPX_FORCEINLINE void wait_all(std::vector<Future>&& values)
      ^

/dss/dsshome1/lrz/sys/compilers/gcc/8.2.0/bin/../include/c++/8.2.0/atomic(187): error: static assertion failed with "std::atomic requires a trivially copyable type"
        static_assert(__is_trivially_copyable(_Tp),
        ^
          detected during:
            instantiation of class "std::atomic<_Tp> [with _Tp=boost::lockfree::detail::tagged_ptr<boost::lockfree::deque_node<hpx::threads::thread_data *>>]" at line 53 of "/home/hpc/pr63so/ga84zod2/hpx/hpx/util/lockfree/deque.hpp"
            instantiation of class "boost::lockfree::deque_node<T> [with T=hpx::threads::thread_data *]" at line 151 of "/home/hpc/pr63so/ga84zod2/hpx/hpx/util/lockfree/deque.hpp"
            instantiation of class "boost::lockfree::deque<T, freelist_t, Alloc> [with T=hpx::threads::thread_data *, freelist_t=boost::lockfree::caching_freelist_t, Alloc=std::allocator<hpx::threads::thread_data *>]" at line 57 of "/home/hpc/pr63so/ga84zod2/hpx/hpx/runtime/threads/policies/lockfree_queue_backends.hpp"
            instantiation of class "hpx::threads::policies::lockfree_fifo_backend<T> [with T=hpx::threads::thread_data *]" at line 1153 of "/home/hpc/pr63so/ga84zod2/hpx/hpx/runtime/threads/policies/thread_queue.hpp"
            instantiation of class "hpx::threads::policies::thread_queue<Mutex, PendingQueuing, StagedQueuing, TerminatedQueuing> [with Mutex=hpx::compat::mutex, PendingQueuing=hpx::threads::policies::lockfree_fifo, StagedQueuing=hpx::threads::policies::lockfree_fifo, TerminatedQueuing=hpx::threads::policies::lockfree_lifo]" at line 1288 of "/home/hpc/pr63so/ga84zod2/hpx/hpx/runtime/threads/policies/local_priority_queue_scheduler.hpp"
            instantiation of class "hpx::threads::policies::local_priority_queue_scheduler<Mutex, PendingQueuing, StagedQueuing, TerminatedQueuing> [with Mutex=hpx::compat::mutex, PendingQueuing=hpx::threads::policies::lockfree_fifo, StagedQueuing=hpx::threads::policies::lockfree_fifo, TerminatedQueuing=hpx::threads::policies::lockfree_lifo]" at line 43 of "/home/hpc/pr63so/ga84zod2/hpx/hpx/runtime/threads/policies/static_priority_queue_scheduler.hpp"
            instantiation of class "hpx::threads::policies::static_priority_queue_scheduler<Mutex, PendingQueuing, StagedQueuing, TerminatedQueuing> [with Mutex=hpx::compat::mutex, PendingQueuing=hpx::threads::policies::lockfree_fifo, StagedQueuing=hpx::threads::policies::lockfree_fifo, TerminatedQueuing=hpx::threads::policies::lockfree_lifo]" at line 116 of "/home/hpc/pr63so/ga84zod2/hpx/hpx/runtime/threads/executors/this_thread_executors.hpp"
            instantiation of class "hpx::threads::executors::detail::this_thread_executor<Scheduler> [with Scheduler=hpx::threads::policies::static_priority_queue_scheduler<hpx::compat::mutex, hpx::threads::policies::lockfree_fifo, hpx::threads::policies::lockfree_fifo, hpx::threads::policies::lockfree_lifo>]" at line 483 of "/home/hpc/pr63so/ga84zod2/hpx/src/runtime/threads/executors/this_thread_executors.cpp"

/dss/dsshome1/lrz/sys/compilers/gcc/8.2.0/bin/../include/c++/8.2.0/atomic(187): error: static assertion failed with "std::atomic requires a trivially copyable type"
        static_assert(__is_trivially_copyable(_Tp),
        ^
          detected during:
            instantiation of class "std::atomic<_Tp> [with _Tp=boost::lockfree::detail::tagged_ptr<boost::lockfree::deque_node<hpx::util::tuple<hpx::threads::thread_init_data, hpx::threads::thread_state_enum> *>>]" at line 53 of "/home/hpc/pr63so/ga84zod2/hpx/hpx/util/lockfree/deque.hpp"
            instantiation of class "boost::lockfree::deque_node<T> [with T=hpx::util::tuple<hpx::threads::thread_init_data, hpx::threads::thread_state_enum> *]" at line 151 of "/home/hpc/pr63so/ga84zod2/hpx/hpx/util/lockfree/deque.hpp"
            instantiation of class "boost::lockfree::deque<T, freelist_t, Alloc> [with T=hpx::util::tuple<hpx::threads::thread_init_data, hpx::threads::thread_state_enum> *, freelist_t=boost::lockfree::caching_freelist_t, Alloc=std::allocator<hpx::util::tuple<hpx::threads::thread_init_data, hpx::threads::thread_state_enum> *>]" at line 57 of "/home/hpc/pr63so/ga84zod2/hpx/hpx/runtime/threads/policies/lockfree_queue_backends.hpp"
            instantiation of class "hpx::threads::policies::lockfree_fifo_backend<T> [with T=hpx::util::tuple<hpx::threads::thread_init_data, hpx::threads::thread_state_enum> *]" at line 1164 of "/home/hpc/pr63so/ga84zod2/hpx/hpx/runtime/threads/policies/thread_queue.hpp"
            instantiation of class "hpx::threads::policies::thread_queue<Mutex, PendingQueuing, StagedQueuing, TerminatedQueuing> [with Mutex=hpx::compat::mutex, PendingQueuing=hpx::threads::policies::lockfree_fifo, StagedQueuing=hpx::threads::policies::lockfree_fifo, TerminatedQueuing=hpx::threads::policies::lockfree_lifo]" at line 1288 of "/home/hpc/pr63so/ga84zod2/hpx/hpx/runtime/threads/policies/local_priority_queue_scheduler.hpp"
            instantiation of class "hpx::threads::policies::local_priority_queue_scheduler<Mutex, PendingQueuing, StagedQueuing, TerminatedQueuing> [with Mutex=hpx::compat::mutex, PendingQueuing=hpx::threads::policies::lockfree_fifo, StagedQueuing=hpx::threads::policies::lockfree_fifo, TerminatedQueuing=hpx::threads::policies::lockfree_lifo]" at line 43 of "/home/hpc/pr63so/ga84zod2/hpx/hpx/runtime/threads/policies/static_priority_queue_scheduler.hpp"
            instantiation of class "hpx::threads::policies::static_priority_queue_scheduler<Mutex, PendingQueuing, StagedQueuing, TerminatedQueuing> [with Mutex=hpx::compat::mutex, PendingQueuing=hpx::threads::policies::lockfree_fifo, StagedQueuing=hpx::threads::policies::lockfree_fifo, TerminatedQueuing=hpx::threads::policies::lockfree_lifo]" at line 116 of "/home/hpc/pr63so/ga84zod2/hpx/hpx/runtime/threads/executors/this_thread_executors.hpp"
            instantiation of class "hpx::threads::executors::detail::this_thread_executor<Scheduler> [with Scheduler=hpx::threads::policies::static_priority_queue_scheduler<hpx::compat::mutex, hpx::threads::policies::lockfree_fifo, hpx::threads::policies::lockfree_fifo, hpx::threads::policies::lockfree_lifo>]" at line 483 of "/home/hpc/pr63so/ga84zod2/hpx/src/runtime/threads/executors/this_thread_executors.cpp"

compilation aborted for /home/hpc/pr63so/ga84zod2/hpx/src/runtime/threads/detail/scheduled_thread_pool.cpp (code 2)
src/CMakeFiles/hpx.dir/build.make:1622: recipe for target 'src/CMakeFiles/hpx.dir/runtime/threads/detail/scheduled_thread_pool.cpp.o' failed
make[2]: *** [src/CMakeFiles/hpx.dir/runtime/threads/detail/scheduled_thread_pool.cpp.o] Error 2
make[2]: *** Waiting for unfinished jobs....
In file included from /home/hpc/pr63so/ga84zod2/hpx/hpx/parallel/executors/execution.hpp(21),
                 from /home/hpc/pr63so/ga84zod2/hpx/hpx/lcos/local/packaged_continuation.hpp(23),
                 from /home/hpc/pr63so/ga84zod2/hpx/hpx/lcos/future.hpp(1715),
                 from /home/hpc/pr63so/ga84zod2/hpx/hpx/runtime/threads/thread_pool_base.hpp(16),
                 from /home/hpc/pr63so/ga84zod2/hpx/hpx/runtime/threads/policies/scheduler_base.hpp(15),
                 from /home/hpc/pr63so/ga84zod2/hpx/hpx/runtime/threads/policies/local_queue_scheduler.hpp(15),
                 from /home/hpc/pr63so/ga84zod2/hpx/src/runtime/threads/executors/thread_pool_attached_executors.cpp(10):
/home/hpc/pr63so/ga84zod2/hpx/hpx/lcos/wait_all.hpp(382): warning #3058: GNU attributes on a template redeclaration have no effect
      HPX_FORCEINLINE void wait_all(std::vector<Future>&& values)
      ^

remark #11074: Inlining inhibited by limit max-size 
remark #11074: Inlining inhibited by limit max-total-size 
remark #11076: To get full report use -qopt-report=4 -qopt-report-phase ipo
compilation aborted for /home/hpc/pr63so/ga84zod2/hpx/src/runtime/threads/executors/this_thread_executors.cpp (code 2)
src/CMakeFiles/hpx.dir/build.make:1886: recipe for target 'src/CMakeFiles/hpx.dir/runtime/threads/executors/this_thread_executors.cpp.o' failed
make[2]: *** [src/CMakeFiles/hpx.dir/runtime/threads/executors/this_thread_executors.cpp.o] Error 2
In file included from /home/hpc/pr63so/ga84zod2/hpx/hpx/parallel/executors/execution.hpp(21),
                 from /home/hpc/pr63so/ga84zod2/hpx/hpx/lcos/local/packaged_continuation.hpp(23),
                 from /home/hpc/pr63so/ga84zod2/hpx/hpx/lcos/future.hpp(1715),
                 from /home/hpc/pr63so/ga84zod2/hpx/hpx/runtime/threads/thread_pool_base.hpp(16),
                 from /home/hpc/pr63so/ga84zod2/hpx/hpx/runtime/threads/policies/scheduler_base.hpp(15),
                 from /home/hpc/pr63so/ga84zod2/hpx/hpx/runtime/threads/policies/local_queue_scheduler.hpp(15),
                 from /home/hpc/pr63so/ga84zod2/hpx/src/runtime/threads/executors/thread_pool_executors.cpp(10):
/home/hpc/pr63so/ga84zod2/hpx/hpx/lcos/wait_all.hpp(382): warning #3058: GNU attributes on a template redeclaration have no effect
      HPX_FORCEINLINE void wait_all(std::vector<Future>&& values)
      ^

/dss/dsshome1/lrz/sys/compilers/gcc/8.2.0/bin/../include/c++/8.2.0/atomic(187): error: static assertion failed with "std::atomic requires a trivially copyable type"
        static_assert(__is_trivially_copyable(_Tp),
        ^
          detected during:
            instantiation of class "std::atomic<_Tp> [with _Tp=boost::lockfree::detail::tagged_ptr<boost::lockfree::deque_node<hpx::threads::thread_data *>>]" at line 53 of "/home/hpc/pr63so/ga84zod2/hpx/hpx/util/lockfree/deque.hpp"
            instantiation of class "boost::lockfree::deque_node<T> [with T=hpx::threads::thread_data *]" at line 151 of "/home/hpc/pr63so/ga84zod2/hpx/hpx/util/lockfree/deque.hpp"
            instantiation of class "boost::lockfree::deque<T, freelist_t, Alloc> [with T=hpx::threads::thread_data *, freelist_t=boost::lockfree::caching_freelist_t, Alloc=std::allocator<hpx::threads::thread_data *>]" at line 57 of "/home/hpc/pr63so/ga84zod2/hpx/hpx/runtime/threads/policies/lockfree_queue_backends.hpp"
            instantiation of class "hpx::threads::policies::lockfree_fifo_backend<T> [with T=hpx::threads::thread_data *]" at line 1153 of "/home/hpc/pr63so/ga84zod2/hpx/hpx/runtime/threads/policies/thread_queue.hpp"
            instantiation of class "hpx::threads::policies::thread_queue<Mutex, PendingQueuing, StagedQueuing, TerminatedQueuing> [with Mutex=hpx::compat::mutex, PendingQueuing=hpx::threads::policies::lockfree_fifo, StagedQueuing=hpx::threads::policies::lockfree_fifo, TerminatedQueuing=hpx::threads::policies::lockfree_lifo]" at line 1288 of "/home/hpc/pr63so/ga84zod2/hpx/hpx/runtime/threads/policies/local_priority_queue_scheduler.hpp"
            instantiation of class "hpx::threads::policies::local_priority_queue_scheduler<Mutex, PendingQueuing, StagedQueuing, TerminatedQueuing> [with Mutex=hpx::compat::mutex, PendingQueuing=hpx::threads::policies::lockfree_fifo, StagedQueuing=hpx::threads::policies::lockfree_fifo, TerminatedQueuing=hpx::threads::policies::lockfree_lifo]" at line 127 of "/home/hpc/pr63so/ga84zod2/hpx/hpx/runtime/threads/executors/thread_pool_executors.hpp"
            instantiation of class "hpx::threads::executors::detail::thread_pool_executor<Scheduler> [with Scheduler=hpx::threads::policies::local_priority_queue_scheduler<hpx::compat::mutex, hpx::threads::policies::lockfree_fifo, hpx::threads::policies::lockfree_fifo, hpx::threads::policies::lockfree_lifo>]" at line 500 of "/home/hpc/pr63so/ga84zod2/hpx/src/runtime/threads/executors/thread_pool_executors.cpp"

/dss/dsshome1/lrz/sys/compilers/gcc/8.2.0/bin/../include/c++/8.2.0/atomic(187): error: static assertion failed with "std::atomic requires a trivially copyable type"
        static_assert(__is_trivially_copyable(_Tp),
        ^
          detected during:
            instantiation of class "std::atomic<_Tp> [with _Tp=boost::lockfree::detail::tagged_ptr<boost::lockfree::deque_node<hpx::util::tuple<hpx::threads::thread_init_data, hpx::threads::thread_state_enum> *>>]" at line 53 of "/home/hpc/pr63so/ga84zod2/hpx/hpx/util/lockfree/deque.hpp"
            instantiation of class "boost::lockfree::deque_node<T> [with T=hpx::util::tuple<hpx::threads::thread_init_data, hpx::threads::thread_state_enum> *]" at line 151 of "/home/hpc/pr63so/ga84zod2/hpx/hpx/util/lockfree/deque.hpp"
            instantiation of class "boost::lockfree::deque<T, freelist_t, Alloc> [with T=hpx::util::tuple<hpx::threads::thread_init_data, hpx::threads::thread_state_enum> *, freelist_t=boost::lockfree::caching_freelist_t, Alloc=std::allocator<hpx::util::tuple<hpx::threads::thread_init_data, hpx::threads::thread_state_enum> *>]" at line 57 of "/home/hpc/pr63so/ga84zod2/hpx/hpx/runtime/threads/policies/lockfree_queue_backends.hpp"
            instantiation of class "hpx::threads::policies::lockfree_fifo_backend<T> [with T=hpx::util::tuple<hpx::threads::thread_init_data, hpx::threads::thread_state_enum> *]" at line 1164 of "/home/hpc/pr63so/ga84zod2/hpx/hpx/runtime/threads/policies/thread_queue.hpp"
            instantiation of class "hpx::threads::policies::thread_queue<Mutex, PendingQueuing, StagedQueuing, TerminatedQueuing> [with Mutex=hpx::compat::mutex, PendingQueuing=hpx::threads::policies::lockfree_fifo, StagedQueuing=hpx::threads::policies::lockfree_fifo, TerminatedQueuing=hpx::threads::policies::lockfree_lifo]" at line 1288 of "/home/hpc/pr63so/ga84zod2/hpx/hpx/runtime/threads/policies/local_priority_queue_scheduler.hpp"
            instantiation of class "hpx::threads::policies::local_priority_queue_scheduler<Mutex, PendingQueuing, StagedQueuing, TerminatedQueuing> [with Mutex=hpx::compat::mutex, PendingQueuing=hpx::threads::policies::lockfree_fifo, StagedQueuing=hpx::threads::policies::lockfree_fifo, TerminatedQueuing=hpx::threads::policies::lockfree_lifo]" at line 127 of "/home/hpc/pr63so/ga84zod2/hpx/hpx/runtime/threads/executors/thread_pool_executors.hpp"
            instantiation of class "hpx::threads::executors::detail::thread_pool_executor<Scheduler> [with Scheduler=hpx::threads::policies::local_priority_queue_scheduler<hpx::compat::mutex, hpx::threads::policies::lockfree_fifo, hpx::threads::policies::lockfree_fifo, hpx::threads::policies::lockfree_lifo>]" at line 500 of "/home/hpc/pr63so/ga84zod2/hpx/src/runtime/threads/executors/thread_pool_executors.cpp"

compilation aborted for /home/hpc/pr63so/ga84zod2/hpx/src/runtime/threads/executors/thread_pool_executors.cpp (code 2)
src/CMakeFiles/hpx.dir/build.make:1934: recipe for target 'src/CMakeFiles/hpx.dir/runtime/threads/executors/thread_pool_executors.cpp.o' failed
make[2]: *** [src/CMakeFiles/hpx.dir/runtime/threads/executors/thread_pool_executors.cpp.o] Error 2


```


## Steps to Reproduce the Problem
I am using following configuration:
```
cmake $HOME/hpx/ \
      -DBOOST_ROOT=$BOOST_BASE \
      -DHPX_WITH_MALLOC=system \
      -DHWLOC_ROOT=$HWLOC_BASE \
      -DCMAKE_CXX_FLAGS="-gxx-name=$GCC_BASE/bin/g++ -use-intel-optimized-headers" \
      -DHPX_WITH_CXX11=ON \
      -DHPX_WITH_PARCELPORT_MPI=ON \
      -DCMAKE_C_COMPILER=icc \
      -DCMAKE_CXX_COMPILER=icpc \
      -DMPI_C_COMPILER=mpicc \
      -DMPI_CXX_COMPILER=mpicxx \
      -DCMAKE_INSTALL_PREFIX=~/hpx_icc_mpi

```

## Specifications

 - HPX Version: 1.3.0.
 -Tried with Intel 17.0,18.0,19.0 in combination with GCC 5.0,6.0,7.0,8.0
 -mpi.intel/2017/2018/2019
 -boost/1.61.0-intel
- hwloc/1.11

Thanks for the fast reply!
With the stable tag checkod out I get following error:
```
[ 22%] Building CXX object libs/resource_partitioner/CMakeFiles/hpx_resource_partitioner.dir/src/partitioner.cpp.o
In file included from /gpfs/scratch/pr63so/ga84zod2/hpx/libs/resource_partitioner/../../hpx/runtime/threads/register_thread.hpp(17),
                 from /gpfs/scratch/pr63so/ga84zod2/hpx/libs/resource_partitioner/../../hpx/runtime/threads/thread_helpers.hpp(20),
                 from /gpfs/scratch/pr63so/ga84zod2/hpx/libs/resource_partitioner/../../hpx/lcos/local/spinlock.hpp(20),
                 from /gpfs/scratch/pr63so/ga84zod2/hpx/libs/resource_partitioner/include/hpx/resource_partitioner/detail/partitioner.hpp(13),
                 from /gpfs/scratch/pr63so/ga84zod2/hpx/libs/resource_partitioner/src/partitioner.cpp(8):
/gpfs/scratch/pr63so/ga84zod2/hpx/libs/resource_partitioner/../../hpx/runtime/threads/thread_data.hpp(608): error #3377: constexpr function return is non-constant
          return static_cast<thread_data*>(tid.get());
                 ^

In file included from /gpfs/scratch/pr63so/ga84zod2/hpx/libs/resource_partitioner/../../hpx/runtime/threads/register_thread.hpp(17),
                 from /gpfs/scratch/pr63so/ga84zod2/hpx/libs/resource_partitioner/../../hpx/runtime/threads/thread_helpers.hpp(20),
                 from /gpfs/scratch/pr63so/ga84zod2/hpx/libs/resource_partitioner/../../hpx/lcos/local/spinlock.hpp(20),
                 from /gpfs/scratch/pr63so/ga84zod2/hpx/libs/resource_partitioner/include/hpx/resource_partitioner/detail/partitioner.hpp(13),
                 from /gpfs/scratch/pr63so/ga84zod2/hpx/libs/resource_partitioner/src/detail_partitioner.cpp(12):
/gpfs/scratch/pr63so/ga84zod2/hpx/libs/resource_partitioner/../../hpx/runtime/threads/thread_data.hpp(608): error #3377: constexpr function return is non-constant
          return static_cast<thread_data*>(tid.get());
                 ^

compilation aborted for /gpfs/scratch/pr63so/ga84zod2/hpx/libs/resource_partitioner/src/partitioner.cpp (code 2)
libs/resource_partitioner/CMakeFiles/hpx_resource_partitioner.dir/build.make:86: recipe for target 'libs/resource_partitioner/CMakeFiles/hpx_resource_partitioner.dir/src/partitioner.cpp.o' failed
make[2]: *** [libs/resource_partitioner/CMakeFiles/hpx_resource_partitioner.dir/src/partitioner.cpp.o] Error 2
make[2]: *** Waiting for unfinished jobs....
compilation aborted for /gpfs/scratch/pr63so/ga84zod2/hpx/libs/resource_partitioner/src/detail_partitioner.cpp (code 2)
libs/resource_partitioner/CMakeFiles/hpx_resource_partitioner.dir/build.make:62: recipe for target 'libs/resource_partitioner/CMakeFiles/hpx_resource_partitioner.dir/src/detail_partitioner.cpp.o' failed
make[2]: *** [libs/resource_partitioner/CMakeFiles/hpx_resource_partitioner.dir/src/detail_partitioner.cpp.o] Error 2
CMakeFiles/Makefile2:35104: recipe for target 'libs/resource_partitioner/CMakeFiles/hpx_resource_partitioner.dir/all' failed
make[1]: *** [libs/resource_partitioner/CMakeFiles/hpx_resource_partitioner.dir/all] Error 2
Makefile:138: recipe for target 'all' failed
make: *** [all] Error 2

```

--- a/hpx/runtime/actions/lambda_to_action.hpp
+++ b/hpx/runtime/actions/lambda_to_action.hpp
@@ -96,18 +96,13 @@ namespace hpx { namespace actions
         };
     }
 
-    template<typename F>
+    template <typename F>
     auto lambda_to_action(F&& f)
-    -> decltype( hpx::actions::detail::action_maker() += true
-        ? nullptr
-        : hpx::actions::detail::addr_add() +  f)
+        -> decltype(hpx::actions::detail::action_maker() +=
+            true ? nullptr : hpx::actions::detail::addr_add() + f)
     {
-        HPX_CONSTEXPR auto act =
-            hpx::actions::detail::action_maker() += true
-            ? nullptr
-            : hpx::actions::detail::addr_add() +  f;
-
-        return act;
+        return hpx::actions::detail::action_maker() +=
+            true ? nullptr : hpx::actions::detail::addr_add() + f;
     }
 }}
 

--- a/hpx/runtime/threads/thread_data.hpp
+++ b/hpx/runtime/threads/thread_data.hpp
@@ -603,7 +603,8 @@ namespace hpx { namespace threads {
         bool is_stackless_;
     };
 
-    HPX_CONSTEXPR thread_data* get_thread_id_data(thread_id_type const& tid)
+    HPX_CXX14_CONSTEXPR inline thread_data* get_thread_id_data(
+        thread_id_type const& tid)
     {
         return static_cast<thread_data*>(tid.get());
     }

--- a/libs/config/include/hpx/config/attributes.hpp
+++ b/libs/config/include/hpx/config/attributes.hpp
@@ -78,7 +78,8 @@
 // handle [[fallthrough]]
 #if defined(HPX_HAVE_CXX17_FALLTHROUGH_ATTRIBUTE)
 #   define HPX_FALLTHROUGH [[fallthrough]]
-#elif defined(HPX_GCC_VERSION) && (HPX_GCC_VERSION >= 70000)
+#elif defined(HPX_GCC_VERSION) && (HPX_GCC_VERSION >= 70000) && \
+     !defined(HPX_INTEL_VERSION)
 #   define HPX_FALLTHROUGH __attribute__((fallthrough))
 // All versions of clang supported by HPX have the [[clang::fallthrough]]
 // attribute.

--- a/libs/resource_partitioner/examples/CMakeLists.txt
+++ b/libs/resource_partitioner/examples/CMakeLists.txt
@@ -29,8 +29,14 @@ if(HPX_WITH_MAX_CPU_COUNT AND HPX_WITH_SHARED_PRIORITY_SCHEDULER)
   if (HPX_WITH_CXX14_RETURN_TYPE_DEDUCTION)
     set(example_programs ${example_programs}
       async_customization
-      guided_pool_test
     )
+
+    # Intel compilers have trouble compiling this example
+    if(NOT ${CMAKE_CXX_COMPILER_ID} STREQUAL "Intel")
+      set(example_programs ${example_programs}
+        guided_pool_test
+      )
+    endif()
   endif()
 endif()
 

--- a/libs/serialization/include/hpx/serialization/detail/extra_archive_data.hpp
+++ b/libs/serialization/include/hpx/serialization/detail/extra_archive_data.hpp
@@ -38,7 +38,7 @@ namespace hpx { namespace serialization { namespace detail {
         inline T* get() noexcept;
 
         std::unique_ptr<extra_archive_data_member_base> ptr_;
-        void (*id_)();
+        void (*id_)() noexcept;
     };
 
     struct extra_archive_data_member_base

